src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,7 +24,7 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        this.setValue(address);
```

Comment:
think how you can use a call to the super constructor to set the values, and in addition do the validation in the parent class.